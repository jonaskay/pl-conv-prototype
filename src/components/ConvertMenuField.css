.ConvertMenuField {
  align-items: center;
  display: flex;
  justify-content: space-between;
  margin-top: 1.5rem;
}

.ConvertMenuField-button {
  background: transparent;
  border: 1px solid #FF8A80;
  border-radius: .25rem;
  box-shadow: 
    0 1px 3px 0 rgba(0,0,0,.2), 0 1px 1px 0 rgba(0,0,0,.14), 0 2px 1px -1px rgba(0,0,0,.12);
  color: #FF8A80;
  cursor: pointer;
  font-size: .875rem;
  padding: .5rem .75rem;
}
