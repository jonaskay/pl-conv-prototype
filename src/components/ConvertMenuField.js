import React from 'react';
import './ConvertMenuField.css';

class ConvertMenuField extends React.Component {  
  render() {
    const name = this.props.name;

    return (
      <div className="ConvertMenuField">
        <div>
          <input 
            type="checkbox" 
            className="ConvertMenuField-input"
            checked={this.props.checked} 
            disabled={!this.props.authenticated}
            onClick={this.props.onInputClick} />
          <label style={{color: 'rgba(255,255,255,.5)', marginLeft: '.5rem'}}>{name}</label>
        </div>
        <div>
          {!this.props.authenticated &&
            <button 
              className="ConvertMenuField-button" 
              onClick={this.props.onButtonClick} 
              disabled={this.props.authDisabled}>
              Connect with {name}
            </button>
          }
        </div>
      </div>
    );
  }
}

export default ConvertMenuField;
