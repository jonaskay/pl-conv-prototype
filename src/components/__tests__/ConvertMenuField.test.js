import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ConvertMenuField from '../ConvertMenuField';

Enzyme.configure({ adapter: new Adapter() });

const mockFunc = jest.fn();
const shallowed = shallow(
  <ConvertMenuField name="Foobar" onInputClick={mockFunc} onButtonClick={mockFunc} />
);

it('renders a field', () => {
  expect(shallowed).toMatchSnapshot();
});

it('renders an authenticated field', () => {
  const component = shallow(<ConvertMenuField name="Foobar" authenticated={true} />);
  expect(component).toMatchSnapshot();  
})

it('renders a disabled button', () => {
  const component = shallow(<ConvertMenuField name="Foobar" authDisabled={true} />);
  expect(component).toMatchSnapshot();  
});

it('handles a button click', () => {
  shallowed.find('.ConvertMenuField-button').simulate('click');
  expect(mockFunc).toBeCalled();
});

it('handles a checkbox click', () => {
  shallowed.find('.ConvertMenuField-input').simulate('click');
  expect(mockFunc).toBeCalled();
})
