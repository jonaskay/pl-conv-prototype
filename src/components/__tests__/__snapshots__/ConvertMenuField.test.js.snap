// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a disabled button 1`] = `
<div
  className="ConvertMenuField"
>
  <div>
    <input
      className="ConvertMenuField-input"
      disabled={true}
      type="checkbox"
    />
    <label
      style={
        Object {
          "color": "rgba(255,255,255,.5)",
          "marginLeft": ".5rem",
        }
      }
    >
      Foobar
    </label>
  </div>
  <div>
    <button
      className="ConvertMenuField-button"
      disabled={true}
    >
      Connect with 
      Foobar
    </button>
  </div>
</div>
`;

exports[`renders a field 1`] = `
<div
  className="ConvertMenuField"
>
  <div>
    <input
      className="ConvertMenuField-input"
      disabled={true}
      onClick={[Function]}
      type="checkbox"
    />
    <label
      style={
        Object {
          "color": "rgba(255,255,255,.5)",
          "marginLeft": ".5rem",
        }
      }
    >
      Foobar
    </label>
  </div>
  <div>
    <button
      className="ConvertMenuField-button"
      onClick={[Function]}
    >
      Connect with 
      Foobar
    </button>
  </div>
</div>
`;

exports[`renders an authenticated field 1`] = `
<div
  className="ConvertMenuField"
>
  <div>
    <input
      className="ConvertMenuField-input"
      disabled={false}
      type="checkbox"
    />
    <label
      style={
        Object {
          "color": "rgba(255,255,255,.5)",
          "marginLeft": ".5rem",
        }
      }
    >
      Foobar
    </label>
  </div>
  <div />
</div>
`;
